src/linux/component.rs

```diff
@@ -9,152 +14,392 @@ use std::path::{Path, PathBuf};
+    temperature: Option<f32>,
+    /// Maximum value computed by sysinfo
+    max: Option<f32>,
+    /// Max threshold provided by the chip/kernel previous user write.
+    /// temp[1-*]_max Temperature min value.
+    ///     Unit: millidegree Celsius
+    ///     RW
+    threshold_max: Option<f32>,
+    /// Max threshold provided by the chip/kernel previous user write.
+    /// temp[1-*]_min Temperature min value.
+    ///     Unit: millidegree Celsius
+    ///     RW
+    threshold_min: Option<f32>,
+    /// temp[1-*]_crit	Temperature critical max value, typically greater than
+    /// 	corresponding temp_max values.
+    /// 	Unit: millidegree Celsius
+    /// 	RW
+    threshold_critical: Option<f32>,
+    /// Rare to have that value
+    /// temp[1-*]_type	Sensor type selection.
+    ///	Integers 1 to 6
+    ///	RW
+    ///	1: CPU embedded diode
+    ///	2: 3904 transistor
+    ///	3: thermal diode
+    ///	4: thermistor
+    ///	5: AMD AMDSI
+    ///	6: Intel PECI
+    ///	Not all types are supported by all chips
+    sensor_type: Option<TermalSensorType>,
+    /// May be empty!
+    /// temp[1-*]_label	Suggested temperature channel label.
+    /// Text string
+    /// Should only be created if the driver has hints about what
+    /// this temperature channel is being used for, and user-space
+    /// doesn't. In all other cases, the label is provided by
+    /// user-space.
+    /// RO
     label: String,
+    // TODO:
+    // temp[1-*]_lowest
+    // Historical minimum temperature
+    // Unit: millidegree Celsius
+    // RO
+    //
+    // temp[1-*]_highest
+    // Historical maximum temperature
+    // Unit: millidegree Celsius
+    //
+    // temp[1-*]_lcrit	Temperature critical min value, typically lower than
+    // corresponding temp_min values.
+    // Unit: millidegree Celsius
+    // RW
+    //
+    // temp[1-*]_emergency
+    // 	Temperature emergency max value, for chips supporting more than
+    // 	two upper temperature limits. Must be equal or greater than
+    // 	corresponding temp_crit values.
+    // 	Unit: millidegree Celsius
+    // 	RW
+    /// temp[1-*]_input file to read current temperature.
+    /// It may be absent but we don't continue if absent.
     input_file: PathBuf,
+    /// temp[1-*]_highest file to read if disponnible highest value.
+    highest_file: Option<PathBuf>,
 }
 
+// Read arbitrary data from sysfs.
 fn get_file_line(file: &Path, capacity: usize) -> Option<String> {
     let mut reader = String::with_capacity(capacity);
-    if let Ok(mut f) = File::open(file) {
-        if f.read_to_string(&mut reader).is_ok() {
-            Some(reader)
-        } else {
-            None
-        }
-    } else {
-        None
-    }
+    let mut f = File::open(file).ok()?;
+    f.read_to_string(&mut reader).ok()?;
+    reader.truncate(reader.trim_end().len());
+    Some(reader)
+}
+
+/// Designed at first for reading an `i32` or `u32` aka `c_long`
+/// from a `/sys/class/hwmon` sysfs file.
+#[inline]
```

Comment:
I don't think this function should be inlined considering it's already quite big.